pkgs/development/libraries/onnxruntime/default.nix

```diff
@@ -1,59 +1,107 @@
 { stdenv
 , lib
 , fetchFromGitHub
+, fetchFromGitLab
 , fetchpatch
 , fetchurl
-, pkg-config
+, Foundation
+, abseil-cpp_202206
 , cmake
-, python3Packages
 , libpng
-, zlib
-, eigen
-, protobuf
-, howard-hinnant-date
 , nlohmann_json
-, boost
-, oneDNN
-, abseil-cpp_202111
-, gtest
-, pythonSupport ? false
 , nsync
-, flatbuffers
+, pkg-config
+, python3Packages
+, re2
+, zlib
+, microsoft-gsl
+, iconv
+, pythonSupport ? true
 }:
 
 # Python Support
 #
 # When enabling Python support a wheel is made and stored in a `dist` output.
 # This wheel is then installed in a separate derivation.
 
-assert pythonSupport -> lib.versionOlder protobuf.version "3.20";
-
 stdenv.mkDerivation rec {
   pname = "onnxruntime";
-  version = "1.13.1";
+  version = "1.15.0";
 
   src = fetchFromGitHub {
     owner = "microsoft";
-    repo = "onnxruntime";
+    repo = pname;
     rev = "v${version}";
-    sha256 = "sha256-paaeq6QeiOzwiibbz0GkYZxEI/V80lvYNYTm6AuyAXQ=";
+    sha256 = "sha256-QSMTx6bOpSvtGn6AoYMMAPzaWjopyKe7rqmsnlxT4tM=";
     fetchSubmodules = true;
   };
 
   patches = [
-    # Use dnnl from nixpkgs instead of submodules
-    (fetchpatch {
-      name = "system-dnnl.patch";
-      url = "https://aur.archlinux.org/cgit/aur.git/plain/system-dnnl.diff?h=python-onnxruntime&id=9c392fb542979981fe0026e0fe3cc361a5f00a36";
-      sha256 = "sha256-+kedzJHLFU1vMbKO9cn8fr+9A5+IxIuiqzOfR2AfJ0k=";
-    })
+    ./cmake-version.patch
   ];
 
+  howard-hinnant-date = fetchFromGitHub {
+    owner = "HowardHinnant";
+    repo = "date";
+    rev = "v2.4.1";
+    sha256 = "sha256-BYL7wxsYRI45l8C3VwxYIIocn5TzJnBtU0UZ9pHwwZw=";
+  };
+
+  eigen = fetchFromGitLab {
+    owner = "libeigen";
+    repo = "eigen";
+    # This commit is not part of master or any release!
```

Comment:
thats not true anymore https://gitlab.com/libeigen/eigen/-/commit/d10b27fe37736d2944630ecd7557cefa95cf87c9

Reply:
Using nixpkgs' eigen 3.4 does not build for me on `aarch64-darwin`. So while the commit may now be merged something else appears to be incompatible. Upstream `onnxruntime` still [uses](https://github.com/microsoft/onnxruntime/tree/v1.15.0/cmake/external) this commit (as a submodule) for the `1.15.0` release. Do you have insights on this @snnn?

In any case: I removed that comment since it is indeed factually incorrect. 

<details>
<summary> Build error using nixpkgs' eigen</summary>

```
In file included from /tmp/nix-build-onnxruntime-1.15.0.drv-0/source/onnxruntime/contrib_ops/cpu/inverse.cc:8:
In file included from /tmp/nix-build-onnxruntime-1.15.0.drv-0/source/onnxruntime/core/util/math_cpuonly.h:68:
In file included from /nix/store/aq0wbcxh58drkasgfr0rkidmmi0liwk0-source/Eigen/Core:358:
/nix/store/aq0wbcxh58drkasgfr0rkidmmi0liwk0-source/Eigen/src/Core/PartialReduxEvaluator.h:57:64: error: no matching function for call to 'pset1'
PacketType packetwise_redux_empty_value(const Func& ) { return pset1<PacketType>(0); }
                                                               ^~~~~~~~~~~~~~~~~
/nix/store/aq0wbcxh58drkasgfr0rkidmmi0liwk0-source/Eigen/src/Core/PartialReduxEvaluator.h:112:14: note: in instantiation of function template specialization 'Eigen::internal::packetwise_redux_empty_value<__attribute__((neon_vector_type(8))) __fp16, Eigen::internal::scalar_sum_op<Eigen::half, Eigen::half>>' requested here
      return packetwise_redux_empty_value<PacketType>(func);
             ^
/nix/store/aq0wbcxh58drkasgfr0rkidmmi0liwk0-source/Eigen/src/Core/PartialReduxEvaluator.h:219:87: note: in instantiation of function template specialization 'Eigen::internal::packetwise_redux_impl<Eigen::internal::scalar_sum_op<Eigen::half, Eigen::half>, Eigen::internal::redux_evaluator<Eigen::Block<const Eigen::CwiseUnaryOp<Eigen::internal::scalar_abs_op<Eigen::half>, const Eigen::Matrix<Eigen::half, -1, -1, 1, -1, -1>>, -1, 8, true>>, 0>::run<__attribute__((neon_vector_type(8))) __fp16>' requested here
    PacketType p = internal::packetwise_redux_impl<BinaryOp,PanelEvaluator>::template run<PacketType>(panel_eval,m_functor.binaryFunc(),m_arg.outerSize());
                                                                                      ^
/nix/store/aq0wbcxh58drkasgfr0rkidmmi0liwk0-source/Eigen/src/Core/Redux.h:251:48: note: in instantiation of function template specialization 'Eigen::internal::evaluator<Eigen::PartialReduxExpr<const Eigen::CwiseUnaryOp<Eigen::internal::scalar_abs_op<Eigen::half>, const Eigen::Matrix<Eigen::half, -1, -1, 1, -1, -1>>, Eigen::internal::member_sum<Eigen::half, Eigen::half>, 0>>::packet<0, __attribute__((neon_vector_type(8))) __fp16>' requested here
      PacketScalar packet_res0 = eval.template packet<alignment,PacketScalar>(alignedStart);
                                               ^
/nix/store/aq0wbcxh58drkasgfr0rkidmmi0liwk0-source/Eigen/src/Core/Redux.h:418:53: note: in instantiation of function template specialization 'Eigen::internal::redux_impl<Eigen::internal::scalar_max_op<Eigen::half, Eigen::half, 0>, Eigen::internal::redux_evaluator<Eigen::PartialReduxExpr<const Eigen::CwiseUnaryOp<Eigen::internal::scalar_abs_op<Eigen::half>, const Eigen::Matrix<Eigen::half, -1, -1, 1, -1, -1>>, Eigen::internal::member_sum<Eigen::half, Eigen::half>, 0>>, 3, 0>::run<Eigen::PartialReduxExpr<const Eigen::CwiseUnaryOp<Eigen::internal::scalar_abs_op<Eigen::half>, const Eigen::Matrix<Eigen::half, -1, -1, 1, -1, -1>>, Eigen::internal::member_sum<Eigen::half, Eigen::half>, 0>>' requested here
  return internal::redux_impl<Func, ThisEvaluator>::run(thisEval, func, derived());
                                                    ^
/nix/store/aq0wbcxh58drkasgfr0rkidmmi0liwk0-source/Eigen/src/Core/Redux.h:448:20: note: in instantiation of function template specialization 'Eigen::DenseBase<Eigen::PartialReduxExpr<const Eigen::CwiseUnaryOp<Eigen::internal::scalar_abs_op<Eigen::half>, const Eigen::Matrix<Eigen::half, -1, -1, 1, -1, -1>>, Eigen::internal::member_sum<Eigen::half, Eigen::half>, 0>>::redux<Eigen::internal::scalar_max_op<Eigen::half, Eigen::half, 0>>' requested here
  return derived().redux(Eigen::internal::scalar_max_op<Scalar,Scalar, NaNPropagation>());
                   ^
/nix/store/aq0wbcxh58drkasgfr0rkidmmi0liwk0-source/Eigen/src/Core/DenseBase.h:466:14: note: (skipping 8 contexts in backtrace; use -ftemplate-backtrace-limit=0 to see all)
      return maxCoeff<PropagateFast>();
             ^
/nix/store/aq0wbcxh58drkasgfr0rkidmmi0liwk0-source/Eigen/src/Core/AssignEvaluator.h:858:3: note: in instantiation of function template specialization 'Eigen::internal::call_assignment_no_alias<Eigen::Map<Eigen::Matrix<Eigen::half, -1, -1, 1, -1, -1>, 0, Eigen::Stride<0, 0>>, Eigen::Inverse<Eigen::Map<const Eigen::Matrix<Eigen::half, -1, -1, 1, -1, -1>, 0, Eigen::Stride<0, 0>>>, Eigen::internal::assign_op<Eigen::half, Eigen::half>>' requested here
  call_assignment_no_alias(dst, src, func);
  ^
/nix/store/aq0wbcxh58drkasgfr0rkidmmi0liwk0-source/Eigen/src/Core/AssignEvaluator.h:836:3: note: in instantiation of function template specialization 'Eigen::internal::call_assignment<Eigen::Map<Eigen::Matrix<Eigen::half, -1, -1, 1, -1, -1>, 0, Eigen::Stride<0, 0>>, Eigen::Inverse<Eigen::Map<const Eigen::Matrix<Eigen::half, -1, -1, 1, -1, -1>, 0, Eigen::Stride<0, 0>>>, Eigen::internal::assign_op<Eigen::half, Eigen::half>>' requested here
  call_assignment(dst, src, internal::assign_op<typename Dst::Scalar,typename Src::Scalar>());
  ^
/nix/store/aq0wbcxh58drkasgfr0rkidmmi0liwk0-source/Eigen/src/Core/Assign.h:66:13: note: in instantiation of function template specialization 'Eigen::internal::call_assignment<Eigen::Map<Eigen::Matrix<Eigen::half, -1, -1, 1, -1, -1>, 0, Eigen::Stride<0, 0>>, Eigen::Inverse<Eigen::Map<const Eigen::Matrix<Eigen::half, -1, -1, 1, -1, -1>, 0, Eigen::Stride<0, 0>>>>' requested here
  internal::call_assignment(derived(), other.derived());
            ^
/nix/store/aq0wbcxh58drkasgfr0rkidmmi0liwk0-source/Eigen/src/Core/Map.h:162:5: note: in instantiation of function template specialization 'Eigen::MatrixBase<Eigen::Map<Eigen::Matrix<Eigen::half, -1, -1, 1, -1, -1>, 0, Eigen::Stride<0, 0>>>::operator=<Eigen::Inverse<Eigen::Map<const Eigen::Matrix<Eigen::half, -1, -1, 1, -1, -1>, 0, Eigen::Stride<0, 0>>>>' requested here
    EIGEN_INHERIT_ASSIGNMENT_OPERATORS(Map)
    ^
/nix/store/aq0wbcxh58drkasgfr0rkidmmi0liwk0-source/Eigen/src/Core/util/Macros.h:1232:5: note: expanded from macro 'EIGEN_INHERIT_ASSIGNMENT_OPERATORS'
    EIGEN_INHERIT_ASSIGNMENT_EQUAL_OPERATOR(Derived) \
    ^
/nix/store/aq0wbcxh58drkasgfr0rkidmmi0liwk0-source/Eigen/src/Core/util/Macros.h:1201:108: note: expanded from macro 'EIGEN_INHERIT_ASSIGNMENT_EQUAL_OPERATOR'
    EIGEN_DEVICE_FUNC EIGEN_STRONG_INLINE Derived& operator=(const DenseBase<OtherDerived>& other) { Base::operator=(other.derived()); return *this; }
                                                                                                           ^
/tmp/nix-build-onnxruntime-1.15.0.drv-0/source/onnxruntime/contrib_ops/cpu/inverse.cc:62:19: note: in instantiation of function template specialization 'Eigen::Map<Eigen::Matrix<Eigen::half, -1, -1, 1, -1, -1>, 0, Eigen::Stride<0, 0>>::operator=<Eigen::Inverse<Eigen::Map<const Eigen::Matrix<Eigen::half, -1, -1, 1, -1, -1>, 0, Eigen::Stride<0, 0>>>>' requested here
    output_matrix = input_matrix.inverse();
                  ^
/nix/store/aq0wbcxh58drkasgfr0rkidmmi0liwk0-source/Eigen/src/Core/GenericPacketMath.h:615:1: note: candidate function template not viable: no known conversion from 'int' to 'const typename unpacket_traits<__attribute__((neon_vector_type(8))) __fp16>::type' (aka 'const Eigen::half') for 1st argument
pset1(const typename unpacket_traits<Packet>::type& a) { return a; }
```
</details>

Reply:
What's the build error? Our latest code uses eigen 3.4.  https://github.com/microsoft/onnxruntime/blob/main/cmake/deps.txt#L52
If there was any problem, it should have been fixed in the main branch.

Reply:
Thanks! The build error is (collapsed) in the above message.

Reply:
What's your gcc version? Could it be too low?

Reply:
This was with gcc 12.2.0. Looking at [ `deps.txt` for 1.15](https://github.com/microsoft/onnxruntime/blob/v1.15.0/cmake/deps.txt) shows that those changes didn't make the cut for the latest release. I think having the out-of-tree version of Eigen is good for now and can be revisited after the next release. Thanks anyhow!